vmdb/app/models/orchestration_template.rb

```diff
@@ -7,30 +7,51 @@ class OrchestrationTemplate < ActiveRecord::Base
 
   has_many :stacks, :class_name => "OrchestrationStack"
 
-  validates_uniqueness_of :md5
+  default_value_for :draft, false
+
+  validates :md5, :uniqueness => {:scope => :draft}, :unless => :draft?
+
+  def self.available
+    where(:draft => false)
+  end
+
+  def self.find_with_content(template_content)
+    available.where(:md5 => calc_md5(template_content)).first
+  end
 
   # Find only by template content. Here we only compare md5 considering the table is expected
   # to be small and the chance of md5 collision is minimal.
   #
   def self.find_or_create_by_contents(hashes)
     hashes = [hashes] unless hashes.kind_of?(Array)
-    md5s = hashes.collect { |hash| Digest::MD5.hexdigest(hash[:content]) }
-    existing_templates = where(:md5 => md5s).index_by(&:md5)
+    md5s = []
+    hashes.delete_if do |hash|
```

Comment:
does the caller expect that you are going to wreck their input?  Does it matter?